templates/pkg/resource/manager_factory.go.tpl

```diff
@@ -66,6 +66,16 @@ func (f *resourceManagerFactory) IsAdoptable() bool {
 	return {{ .CRD.IsAdoptable }}
 }
 
+// GetRequeueOnSuccessSeconds returns true if the resource should be requeued after specified seconds
+// Default is false which means resource will not be requeued after success. 
+func (f *resourceManagerFactory) GetRequeueOnSuccessSeconds() (int, bool) {
+{{- if $reconcileRequeuOnSuccessSeconds := .CRD.ReconcileRequeuOnSuccessSeconds }}
```

Comment:
I do not see any unit tests in this PR. Can you please add them?

Also since "ReconcileRequeuOnSuccessSeconds" returns an int, would not this if statement always be true.
This Unit tests for this code-generation path will validate if my concern is legit.

Reply:
Sure, adding some unit tests.